pkg/authorization/api/v1beta3/types.go

```diff
@@ -113,10 +113,21 @@ type NamedRoleBinding struct {
 type SelfSubjectRulesReview struct {
 	unversioned.TypeMeta `json:",inline"`
 
+	// Spec adds information about how to conduct the check
+	Spec SubjectRulesReviewSpec `json:"status`
+
 	// Status is completed by the server to tell which permissions you have
 	Status SubjectRulesReviewStatus `json:"status,omitempty"`
 }
 
+// SubjectRulesReviewSpec adds information about how to conduct the check
+type SubjectRulesReviewSpec struct {
+	// Scopes to use for the evaluation.  Empty means "use the unscoped (full) permissions of the user/groups".
+	// Nil for a self-SubjectRulesReview, means "use the scopes on this request".
```

Comment:
This doc probably belongs in each type, not in the spec itself.  I.e. don't define the behavior in the struct that is polymorphic